### PR TITLE
 fix(chip group): add prop to allow chip group to be initially expanded

### LINKED ISSUE #1697: PF4: ChipGroup - allow chip group to be expanded via passing prop isOpen
Defaulting isOpen = false is ok, but allow an override to allow the chip group to be initially expanded.

### PR DESCRIPTION
**Pull request by jenny-s51**
**What**: Closes #1697.

### CODE REVIEW

File: packages/patternfly-4/react-core/src/components/ChipGroup/ChipGroup.tsx [hunk 1]
```diff
@@ -11,6 +11,8 @@ export interface ChipGroupProps extends React.HTMLProps<HTMLDivElement> {
   children?: React.ReactNode;
   /** Additional classes added to the chip item */
   className?: string; 
+    /** Flag for having the chip group default to expanded */
+  isOpenOnInit?: boolean;
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/ChipGroup/ChipGroup.tsx]: Hi.  I was just talking to @jschuler  about the name of this prop.  We want to be consistent with the way we name our props across components.  The Page has a similar prop called 'defaultManagedSidebarIsOpen' so we think the chip Group should have a similar name.  How about 'defaultIsOpen`.  I will also rename the Page prop to defaultManagedSidebarIsOpen for consistency.

Comment (jenny-s51) [reply packages/patternfly-4/react-core/src/components/ChipGroup/ChipGroup.tsx]: Sounds good, thank you Titani -- I'll rename it accordingly

**Comment by jenny-s51** [comment]
@tlabaj absolutely!

File: packages/patternfly-4/react-core/src/components/ChipGroup/ChipGroup.tsx [hunk 2]
```diff
@@ -11,6 +11,8 @@ export interface ChipGroupProps extends React.HTMLProps<HTMLDivElement> {
   children?: React.ReactNode;
   /** Additional classes added to the chip item */
   className?: string; 
+    /** Flag for having the chip group default to expanded */
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/ChipGroup/ChipGroup.tsx]: looks like an extra space here

File: packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ChipGroupDemo/ChipGroupDemo.tsx [hunk 3]
```diff
@@ -46,7 +46,7 @@ export class BadgeChip extends Component<{}, BadgeChipState> {
   render() {
     const { badgeChipArray } = this.state;
     return (
-      <ChipGroup>
+      <ChipGroup defaultIsOpen={false}>
```
Comment (tlabaj) [inline packages/patternfly-4/react-integration/demo-app-ts/src/components/demos/ChipGroupDemo/ChipGroupDemo.tsx]: Can you add a demo for the defaultIsOpen being true and a test for that as well.

File: packages/patternfly-4/react-integration/cypress/integration/chipgroupdefaultisopen.spec.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
+describe('Chip Group Demo Test', () => {
+    it('Navigate to demo section', () => {
+      cy.visit('http://localhost:3000/');
+      cy.get('#chipgroup-default-is-open-demo-nav-item-link').click(); 
+      cy.url().should('eq', 'http://localhost:3000/chipgroup-default-is-open-demo-nav-link');
+    });
+  
+    it('Verify chip default text', () => {
```
Comment (tlabaj) [inline packages/patternfly-4/react-integration/cypress/integration/chipgroupdefaultisopen.spec.ts]: shouldn't we be verifying the chip group is open?

Comment (jenny-s51) [reply packages/patternfly-4/react-integration/cypress/integration/chipgroupdefaultisopen.spec.ts]: Yes, nice catch! fixed it 👍

File: packages/patternfly-4/react-core/src/components/ChipGroup/ChipGroup.tsx [hunk 5]
```diff
@@ -29,15 +31,16 @@ export class ChipGroup extends React.Component<ChipGroupProps, ChipGroupState>{
   constructor(props: ChipGroupProps) {
     super(props); 
     this.state = {
-      isOpen: false
+      isOpen: this.props.defaultIsOpen
```
Comment (redallen) [inline packages/patternfly-4/react-core/src/components/ChipGroup/ChipGroup.tsx]: Don't need `this.`, but `super(props)` does set `this.props` so you're fine :)

**Pull request merged**